src/ConnectedMode/BranchMatcher.cs

```diff
@@ -93,7 +95,7 @@ private async Task<string> DoGetMatchingBranch(string projectKey, IRepository gi
                 return null;
             }
 
-            var remoteBranches = await sonarQubeService.GetProjectBranchesAsync(projectKey, token);
+            var remoteBranches = (await sonarQubeService.GetProjectBranchesAsync(projectKey, token)).Where(b => b.Type != shortBranchType);
```

Comment:
Do we still want to ignore it even if it's the exact match?

Reply:
Yes. https://sonarsource.slack.com/archives/C02E6L5C01H/p1711613419949759?thread_ts=1711552233.646179&cid=C02E6L5C01H